core/app_apis_services.go

```diff
@@ -57,6 +58,118 @@ func (app *Application) unsubscribeToTopic(orgID string, appID string, token str
 	return err
 }
 
+func (app *Application) getUserData(orgID string, appID string, userID string) (*model.UserDataResponse, error) {
+	var (
+		usersResponse            []model.UserResponse
+		messageResponse          []model.MessageResponse
+		messageRecipientResponse []model.MessageRecipientResponse
+		queueResponse            []model.QueueItemResponse
+		err                      error
+		mu                       sync.Mutex
+		wg                       sync.WaitGroup
+	)
+
+	// Define an error channel to capture any errors from the goroutines
+	errChan := make(chan error, 4)
+
+	// Fetch messages concurrently
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		messages, e := app.storage.GetMessagesByUserID(userID)
```

Comment:
This will always gives empty result as there are no any records in the database with this filter. The reason is that for sender stays the building block but not the user.